test/unit/test.database.js

```diff
@@ -1,13 +1,247 @@
+    const tableName = tableNames[i];
+    await dumpTable(db, tableName);
+  }
+}
+
+function makeLogFactory(universe) {
+  function common() {
+    return {
+      transactionHash: uuid.v4(),
+      transactionIndex: 0,
+      logIndex: 0,
+      blockNumber: 2,
+      blockHash: uuid.v4(),
+      removed: false,
+      invalid: false,
+      address: "AUGUR_CONTRACT",
+      contractName: "Augur",
+    };
+  }
+
+  return Object.assign({}, {
+    UniverseCreated: args => {
+      return Object.assign(common(), {
+        eventName: "UniverseCreated",
+        parentUniverse: null,  // leave null for root universe
+        childUniverse: universe,  // the universe you're creating
+        payoutNumerators: [],
+
+      }, args || {});
+    },
+    UniverseForked: args => {
+      required(args, "universe");  // forking tests should be explicit about their universes
+      return Object.assign(common(), {
+        eventName: "UniverseForked",
+        universe,
+      }, args || {});
+    },
+    TokensMinted: args => {
+      required(args, "token", "target", "amount", "market");
+      return Object.assign(common(), {
+        eventName: "TokensMinted",
+        universe,
+      }, args || {});
+    },
+    FeeWindowCreated: args => {
+      required(args, "id", "feeWindow", "startTime", "endTime");
+      return Object.assign(common(), {
+        eventName: "FeeWindowCreated",
+        universe,
+      }, args || {});
+    },
+    TokensTransferred: args => {
+      required(args, "token", "from", "to", "value", "market");
+      return Object.assign(common(), {
+        eventName: "TokensTransferred",
+        universe,
+      }, args || {});
+    },
+    MarketCreated: args => {
+      required(args, "market", "marketCreator");
+      return Object.assign(common(), {
+        eventName: "MarketCreated",
+        universe,
+        topic: uuid.v4(),
+        description: uuid.v4(),
+        extraInfo: {
+          resolutionSource: "",
+          tags: [],
+        },
+        marketCreationFee: "0.01",  // how much is extracted from trades as market creator
+        minPrice: "0",
+        maxPrice: "1",
+        // See constants.ts/MarketType enum for details.
+        // 0 -> yesNo aka binary
+        // 1 -> categorical aka 3-8 outcomes. Log must include "outcomes".
+        marketType: "0",
+        outcomes: [],  // if marketType=1 then this must be specified
+      }, args || {});
+    },
+    OrderCreated: args => {
+      required(
+        args,
+        "shareToken",
+        "amount",
+        "price",
+        "creator",
+        "moneyEscrowed",
+        "sharesEscrowed",
+        "tradeGroupId",
+      );
+      return Object.assign(common(), {
+        eventName: "OrderCreated",
+        universe,
+        orderType: "0",
+        orderId: uuid.v4(),
+      }, args || {});
+    },
+  });
+}
+
+function hexify(integer) {
+  return "0x" + (integer).toString(16);
+}
+
+function makeFakeBlock(number, overrides) {
```

Comment:
So there's really just two important bits here: blockNumber and timestamp. I see we could do
`makeFakeBlock(2, {timestamp: 4000});`, but it might be cool if timestamp was a first-class option (since it changes market state and dispute rounds and whatnot)? But this is totally workable, too.